controllers/ip-location.js

```diff
@@ -0,0 +1,41 @@
+/*eslint-disable no-process-env*/
+// using process.env here to keep the banner/VPN vars contained – for when the vpn-banner is eventually removed
+
+// IP location data includes GeoLite2 data created by MaxMind, available from https://www.maxmind.com.
+// Test IPs by comparing to the corresponding json, for example: https://github.com/maxmind/MaxMind-DB/blob/main/source-data/GeoLite2-City-Test.json
+
+"use strict";
+
+require("dotenv").config();
+const maxmindDb = process.env.GEOIP_GEOLITE2_PATH + process.env.GEOIP_GEOLITE2_CITY_FILENAME;
+const Reader = require("@maxmind/geoip2-node").Reader;
+
+async function ipLocation(req, res) {
+  let reader, geoData, countryName, cityName, locationArr;
+  const clientIp = process.env.NODE_ENV === "dev" ? "216.160.83.56" : req.ip; // for dev, return an IP that exists in GeoLite2 test DB, e.g. 216.160.83.56
+
+  if (clientIp === req.session.locationData?.clientIp) return res.status(200).json(req.session.locationData);
+
+  try {
+    reader = await Reader.open(maxmindDb);
```

Comment:
I'm pretty ignorant here (and in general), but what how does this line work?
Is it opening the maxmind DB on _each_ request? Or is this somehow cached?

Reply:
Good idea. We could move this into something like the `LocaleUtils.loadLanguagesIntoApp`, `FormUtils.loadCountriesIntoApp`, `HIBP.loadBreachesIntoApp` so it loads the data into the app a single time during server startup. (they're only 20K and 18K)

Reply:
@groovecoder I assume you're referring to the test MaxMind databases – they are actually stripped down for testing.  I imagine the actual databases are much larger...

@pdehaan I took this block to mean we're caching session data to redis: 
https://github.com/mozilla/blurts-server/blob/1a940301908ecd1fd0d2fb1a28dc8762d4609dfa/server.js#L195-L207

So, if a user's ip address matches an ip stored in session from an earlier successful lookup, we return that data instead of opening/reading the MaxMind database.

Admittedly this is my first work with Redis, so let me know if I've missed something!

Reply:
That's good to optimize each session individually, but we should still load the maxmind DB into the application memory during server startup. Or else 1000 visitors will cause 1000 file open-and-read-into-memory operations when we may only need a single open-and-read-into-memory operation for all visitors.

Reply:
This has been updated – thanks for flagging.  I split the code out into a separate service from the controller.  See changes here: https://github.com/mozilla/blurts-server/pull/2284/commits/52c15b26fb4d04092e51e63459158ea8685f8c3c